lib/colors.ts

```diff
@@ -160,6 +162,43 @@ export function toPercent(value: number): string {
   return `${clean(100 * value)}%`
 }
 
+function isHex(sequence: number): RegExp {
+  return new RegExp(`^#?([\\da-f]{${sequence}})$`, 'i')
+}
+function isHex3(value: string): boolean {
+  return isHex(3).test(value)
+}
+function isHex4(value: string): boolean {
+  return isHex(4).test(value)
+}
+function isHex6(value: string): boolean {
+  return isHex(6).test(value)
+}
+export function isHex8(value: string): boolean {
+  return isHex(8).test(value)
+}
+
+function isHexNotation(value: string): boolean {
+  return [isHex3, isHex4, isHex6, isHex8].some(fn => fn(value))
+}
+
+const OUTPUT_FORMAT_MAP: Partial<Record<Mode, OutputFormats>> = {
+  hsl: 'hsl',
+  lab: 'lab',
+  lch: 'lch',
+  oklab: 'oklab',
+  oklch: 'numbers',
+  p3: 'p3',
+  rgb: 'rgb'
+}
+
+export function toOutputFormat(value: string): OutputFormats | undefined {
+  let parsed = parseAnything(value)
+  if (!parsed) return undefined
+  if (isHexNotation(value)) return 'hex'
```

Comment:
We should use hex/rgba mode for hex input.

Otherwise, on adding alpha, we will show rare hex8 format to user. Most users don’t familiar with it.

Reply:
Done